controllers/registeredservice_controller.go

```diff
@@ -161,12 +167,184 @@ func (r *RegisteredServiceReconciler) addServiceToCatalog(ctx context.Context, s
 	return nil
 }
 
+func (r *RegisteredServiceReconciler) registerHealthcheck(ctx context.Context, rs *primazaiov1alpha1.RegisteredService) error {
+	// Make a CronJob that runs the healthcheck.  The CronJob handler will take
+	// care of the rest.
+	cronjob := batchv1.CronJob{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      rs.Name,
+			Namespace: rs.Namespace,
+		},
+	}
+	operation, err := controllerutil.CreateOrUpdate(ctx, r.Client, &cronjob, func() error {
+		var one int32 = 1
+		var two int32 = 2
+		var thirty int64 = 30
+		var uid int64 = 65530
+		t := true
+		f := false
+		container := corev1.Container{
+			Name:            "healthcheck",
+			Command:         rs.Spec.HealthCheck.Container.Command,
+			Image:           rs.Spec.HealthCheck.Container.Image,
+			ImagePullPolicy: corev1.PullIfNotPresent,
+			SecurityContext: &corev1.SecurityContext{
+				Capabilities: &corev1.Capabilities{
+					Drop: []corev1.Capability{"ALL"},
+				},
+				Privileged:               &f,
+				RunAsNonRoot:             &t,
+				AllowPrivilegeEscalation: &f,
+			},
+		}
+		cronjob.Spec = batchv1.CronJobSpec{
+			// run every n minutes
+			Schedule:                   fmt.Sprintf("*/%d * * * *", rs.Spec.HealthCheck.Container.Minutes),
+			ConcurrencyPolicy:          batchv1.ForbidConcurrent,
+			SuccessfulJobsHistoryLimit: &one,
+			FailedJobsHistoryLimit:     &one,
+			JobTemplate: batchv1.JobTemplateSpec{
+				Spec: batchv1.JobSpec{
+					ActiveDeadlineSeconds: &thirty,
+					BackoffLimit:          &two,
+					Parallelism:           &one,
+					Template: corev1.PodTemplateSpec{
+						Spec: corev1.PodSpec{
+							Containers:    []corev1.Container{container},
+							RestartPolicy: corev1.RestartPolicyNever,
+							SecurityContext: &corev1.PodSecurityContext{
+								RunAsUser:    &uid,
+								RunAsGroup:   &uid,
+								RunAsNonRoot: &t,
+								SeccompProfile: &corev1.SeccompProfile{
+									Type: corev1.SeccompProfileTypeRuntimeDefault,
+								},
+							},
```

Comment:
Hum... How are we passing the secrets needed for the client to authenticate to the service? We are not only checking tcp connectivity user might want to check the actual service is up and running and credentials are accurate to use hence "available" state.

Reply:
I think I'll punt on implementing this for a separate PR.  This is definitely going to take a bit more work, and I'd rather not hold up this PR by implementing it.

Reply:
you got it